src/Microsoft.Android.Build.BaseTasks/AndroidTask.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using Microsoft.Build.Utilities;
+using Xamarin.Build;
+using static System.Threading.Tasks.TaskExtensions;
+
+namespace Microsoft.Android.Build.Tasks
+{
+	// We use this task to ensure that no unhandled exceptions
+	// escape our tasks which would cause an MSB4018
+	public abstract class AndroidTask : Task
+	{
+		public abstract string TaskPrefix { get; }
+
+		public override bool Execute ()
+		{
+			try {
+				return RunTask ();
+			} catch (Exception ex) {
+				Log.LogUnhandledException (TaskPrefix, ex);
+				return false;
+			}
+		}
+
+		public abstract bool RunTask ();
+	}
+
+	public abstract class AndroidAsyncTask : AsyncTask
+	{
+		/// <summary>
+		/// A helper for non-async overrides of RunTaskAsync, etc.
+		/// </summary>
+		public static readonly System.Threading.Tasks.Task Done =
+			System.Threading.Tasks.Task.FromResult (true);
```

Comment:
We can remove this, and use `System.Threading.Tasks.Task.CompletedTask` instead.

I think I added this when I didn't know about `CompletedTask`.

Reply:
Good call. Also do we want to split these up into their own files since we are moving them?

Reply:
I updated this to use CompletedTask and split up all of the Android Task extensions into their own files.